app/org/sagebionetworks/bridge/services/ScheduledActivityService.java

```diff
@@ -366,189 +358,32 @@ private int byteLength(JsonNode node) {
     }
 
     protected List<ScheduledActivity> scheduleActivitiesForPlans(ScheduleContext context) {
-        // Cache compound activity defs, schemas, and surveys to reduce calls from duplicate requests.
-        Map<String, CompoundActivity> compoundActivityCache = new HashMap<>();
-        Map<String, SchemaReference> schemaCache = new HashMap<>();
-        Map<String, SurveyReference> surveyCache = new HashMap<>();
         List<ScheduledActivity> scheduledActivities = new ArrayList<>();
 
         List<SchedulePlan> plans = schedulePlanService.getSchedulePlans(context.getCriteriaContext().getClientInfo(),
                 context.getCriteriaContext().getStudyIdentifier());
-
+        
+        AppConfig appConfig = appConfigService.getAppConfigForUser(context.getCriteriaContext(), false);
+        Map<String, SurveyReference> surveyReferences = (appConfig == null) ? ImmutableMap.of()
+                : Maps.uniqueIndex(appConfig.getSurveyReferences(), SurveyReference::getGuid);
+        Map<String, SchemaReference> schemaReferences = (appConfig == null) ? ImmutableMap.of()
+                : Maps.uniqueIndex(appConfig.getSchemaReferences(), SchemaReference::getId);
```

Comment:
Is it worth adding a minimal test to ScheduledActivityServiceResolveLinksTest with non-null AppConfig and non-empty reference maps, just to make sure we're constructing the maps correctly?

Reply:
Added a test where there's an AppConfig, verify the app configs references are used to resolve the scheduled activities, so the map should be correct.